controllers/volumereplication_controller.go

```diff
@@ -84,6 +87,23 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return ctrl.Result{}, nil
 	}
 
+	// validate the prefix keys in volume replication class
+	parameters, err := removePrefixedParameters(vrcObj.Spec.Parameters)
+	if err != nil {
+		r.Log.Error(err, "failed to remove prefix parameters", "volumeReplicationClass", instance.Spec.VolumeReplicationClass)
```

Comment:
Needs to be rephrased as we are not actually trying to remove prefixed parameters.